core/trino-main/src/main/java/io/trino/event/QueryMonitor.java

```diff
@@ -273,7 +289,138 @@ private QueryMetadata createQueryMetadata(QueryInfo queryInfo, boolean requiresA
                 queryInfo.getSelf(),
                 createTextQueryPlan(queryInfo, anonymizer),
                 createJsonQueryPlan(queryInfo, anonymizer),
-                queryInfo.getOutputStage().flatMap(stage -> stageInfoCodec.toJsonWithLengthLimit(stage, maxJsonLimit)));
+                queryInfo.getOutputStage().flatMap(stage -> stageInfoCodec.toJsonWithLengthLimit(stage, maxJsonLimit)),
+                createStageDetails(queryInfo),
+                createQueryPlanDetails(queryInfo));
```

Comment:
We already provide a query plan in json format in `QueryCompletedEvent.metadata.jsonPlan`, can you use that instead of adding this ?